### ISSUE TITLE
pachd panics on "pachctl run pipeline"

### ISSUE THREAD

**Issue by simeonmiteff** [pachyderm/pachyderm#4049]
**What happened?**:
`pachd` panicked

**What you expected to happen?**:
Pipeline re-run for specific commit specified.

**How to reproduce it (as minimally and precisely as possible)?**:
I run `pachctl run pipeline pipeline-name 993dc824df6b416480a4e20a813f026c`, and I get this:
```
2019-09-04T00:18:50Z INFO pps.API.gopanic {"duration":0.088731527,"request":{"pipeline":{"name":"pipeline-name"},"provenance":[{"commit":{"repo":{"name":"993dc824df6b416480a4e20a813f026c"}}}]},"response":null} []
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1eae3dd]
goroutine 5079 [running]:
github.com/pachyderm/pachyderm/src/server/pps/server.(*apiServer).RunPipeline(0xc000276a20, 0x2941940, 0xc000e33d40, 0xc00158b180, 0x0, 0x0, 0x0)
    src/github.com/pachyderm/pachyderm/src/server/pps/server/api_server.go:2603 +0x4bd
github.com/pachyderm/pachyderm/src/client/pps._API_RunPipeline_Handler.func1(0x2941940, 0xc000e33b30, 0x23cc7c0, 0xc00158b180, 0x23cc7c0, 0xc00158b180, 0x0, 0x0)
    src/github.com/pachyderm/pachyderm/src/client/pps/pps.pb.go:5936 +0x89
github.com/opentracing-contrib/go-grpc.OpenTracingServerInterceptor.func1(0x2941940, 0xc000e33b30, 0x23cc7c0, 0xc00158b180, 0xc0010402c0, 0xc001040300, 0x0, 0x0, 0x28e8d40, 0xc0002163a0)
    src/github.com/pachyderm/pachyderm/vendor/github.com/opentracing-contrib/go-grpc/server.go:44 +0x9db
github.com/pachyderm/pachyderm/src/client/pps._API_RunPipeline_Handler(0x2486760, 0xc000276a20, 0x2941940, 0xc000e33b30, 0xc0011645f0, 0xc000456ea0, 0x2941940, 0xc000e33b30, 0xc000edd130, 0x44)
    src/github.com/pachyderm/pachyderm/src/client/pps/pps.pb.go:5938 +0x158
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000302000, 0x296fbc0, 0xc00157a000, 0xc000f76700, 0xc000ea9a70, 0x3fb14f0, 0x0, 0x0, 0x0)
    src/github.com/pachyderm/pachyderm/vendor/google.golang.org/grpc/server.go:998 +0x470
google.golang.org/grpc.(*Server).handleStream(0xc000302000, 0x296fbc0, 0xc00157a000, 0xc000f76700, 0x0)
    src/github.com/pachyderm/pachyderm/vendor/google.golang.org/grpc/server.go:1278 +0xda6
google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc0010a78a0, 0xc000302000, 0x296fbc0, 0xc00157a000, 0xc000f76700)
    src/github.com/pachyderm/pachyderm/vendor/google.golang.org/grpc/server.go:717 +0x9f
created by google.golang.org/grpc.(*Server).serveStreams.func1
    src/github.com/pachyderm/pachyderm/vendor/google.golang.org/grpc/server.go:715 +0xa1
```
Running the same command again causes another panic.

**Anything else we need to know?**:
I previously got pachyderm into a state where PPS was stuck (pipelines didn't run, and there was no obvious cause - no errors other than the probably-unrelated messages referred to in #4047).

While in this state I tried creating the `edges` pipeline from the quick-start example in the documentation. I found that the new pipeline was in starting/starting state, but no worker pod was being created (no pod at all, not even one with errors). I solved this by deleting the `pachd` pod, and letting it come back up again, at which point pipelines could run again.

Next I fixed an error causing a different pipeline to fail, and updated it with `--reprocess` but the pipeline didn't reprocess, so I decided to use `pachctl run pipeline` to force it, causing the crash.

**Environment?**:

- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.2", GitCommit:"f6278300bebbb750328ac16ee6dd3aa7d3549568", GitTreeState:"clean", BuildDate:"2019-08-05T09:23:26Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"12+", GitVersion:"v1.12.8-gke.10", GitCommit:"f53039cc1e5295eed20969a4f10fb6ad99461e37", GitTreeState:"clean", BuildDate:"2019-06-19T20:48:40Z", GoVersion:"go1.10.8b4", Compiler:"gc", Platform:"linux/amd64"}
```

- Pachyderm CLI and pachd server version (use `pachctl version`):
```
COMPONENT           VERSION             
pachctl             1.9.3               
pachd               1.9.3
```
- Cloud provider (e.g. aws, azure, gke) or local deployment (e.g. minikube vs dockerized k8s):
GKE

- OS (e.g. from /etc/os-release):
```
BUILD_ID=10895.277.0
NAME="Container-Optimized OS"
KERNEL_COMMIT_ID=6b10a0b5b1910970dc3b62c14c011b39343a5279
GOOGLE_CRASH_ID=Lakitu
VERSION_ID=69
BUG_REPORT_URL="https://cloud.google.com/container-optimized-os/docs/resources/support-policy#contact_us"
PRETTY_NAME="Container-Optimized OS from Google"
VERSION=69
GOOGLE_METRICS_PRODUCT_ID=26
HOME_URL="https://cloud.google.com/container-optimized-os/docs"
ID=cos
```
- Others:

**Comment by jdoliner** [comment]
This should be a pretty simple fix, assigning to @adelelopez who worked on `run pipeline` recently.

**Comment by marcadella** [comment]
I got a similar issue except that åpachd did not panic but wasn't accessible for a minute or so. When I could access pachd again the pipeline had not been restarted as requested. Should I open a new issue or this is fixed by #4060 ?
